drivers/virtio/net/virtio_net.c

```diff
@@ -354,10 +378,11 @@ static int virtio_netdev_xmit(struct uk_netdev *dev,
 
 	buf_start = pkt->data;
 	buf_len = pkt->len;
+
 	/**
 	 * Use the preallocated header space for the virtio header.
 	 */
-	rc = uk_netbuf_header(pkt, header_sz);
+	rc = uk_netbuf_header(pkt, VTNET_HDR_SIZE_PADDED(vndev));
```

Comment:
Can we somehow store `VTNET_HDR_SIZE_PADDED(vndev)`/`virtio_net_hdr_size(vndev)` in a separate variable? We are calling at least 3 times (on the base case) `virtio_net_hdr_size()` in `virtio_netdev_xmit()`.

Reply:
Maybe the compiler is smart enough to determine vndev is const inside `virtio_net_hdr_size` and not compute everything multiple times. But I haven't check...

Reply:
I'm almost sure the compiler will optimize that out :)